cfme/configure/settings.py

```diff
@@ -23,109 +25,173 @@ class TimeProfileAddForm(View):
+                description="{} copy".format(self.description),
+                scope=self.scope
+            )
+            changed = view.time_profile_form.fill({
                 'description': "{} copy".format(self.description),
                 'scope': self.scope,
             })
 
         if changed:
-            view.timeprofile_form.save_button.click()
-        return new_timeprofile
+            view.time_profile_form.save_button.click()
+        return new_time_profile
+
+
+@attr.s
+class TimeProfileCollection(BaseCollection):
+
+    ENTITY = TimeProfile
+
+    def create(self, description, scope, days, hours, timezone, cancel=False):
+        """
+        Args:
+            description (str): It's the descriptive name of the time_profile.
+            scope: It's the option 'All User' or 'Current User' from dropdown.
+            days (bool): It's the option to switch on or switch off the days Bootstrap switch.
+            hours (bool): It's the option to swich on or switch off the hours Bootstrap switch.
+            timezone: It's the required Time Zone for the time_profile.
+            cancel (bool) : It's a flag used to cancel or not the create operation.
+
+        return: It returns the object of the newly created time_profile object.
+        """
+
+        time_profile = self.instantiate(description, scope, days, hours, timezone)
 
-    def delete(self):
         view = navigate_to(self, 'All')
-        rows = view.timeprofile_form.table
-        for row in rows:
-            if row.description.text == self.description:
-                row[0].check()
-        view.timeprofile_form.configuration.item_select("Delete selected "
-                                                        "Time Profiles", handle_alert=True)
+        view.time_profile_form.configuration.item_select('Add a new Time Profile')
+        view.time_profile_form.fill({
+            'description': description,
+            'scope': scope,
+            'days': days,
+            'hours': hours,
+            'timezone': timezone,
+        })
+        if not cancel:
+            view.time_profile_form.save_button.click()
+            view.flash.assert_message('Time Profile "{}" was saved'.format(description))
+        return time_profile
+
+    def delete(self, cancel=False, *time_objs):
+        """
+        This method performs the delete operation.
+
+        Args:
+            cancel (bool) : It's a flag used for selecting Ok or Cancel from delete confirmation
+             dialogue box
+            time_objs : It's time profile object.
+        """
+
+        view = navigate_to(self, 'All')
+        rows = view.time_profile_form.table
+        for time_obj in time_objs:
+            for row in rows:
+                if row.description.text == time_obj.description:
+                    row[0].check()
+        if not cancel:
+            view.time_profile_form.configuration.item_select("Delete selected Time Profiles",
```

Comment:
Please collapse this into a single method call, and pass `not cancel` to `handle_alert`